src/pages/login.tsx

```diff
@@ -1,73 +1,121 @@
-        onSubmit={handleSubmit}
-        className="flex max-w-md flex-col gap-4 "
-      >
-        <h2 className="mb-1 text-base font-semibold text-gray-900 dark:text-white">
-          {isSignup ? 'Sign Up' : 'Sign In'}
-        </h2>
-        <div>
-          <div className="mb-2 block">
-            <Label htmlFor="email1" value="Your email" />
-          </div>
-          <TextInput
-            id="email"
-            data-testid="login-email"
-            type="email"
-            placeholder="jane.doe@example.com"
-            required
-          />
-        </div>
-        <div>
-          <div className="mb-2 block">
-            <Label htmlFor="password1" value="Your password" />
-          </div>
-          <TextInput
-            data-testid="login-password"
-            id="password1"
-            type="password"
-            required
-          />
-        </div>
+    <div className="flex flex-col justify-center relative h-full w-1/4 overflow-y-auto m-10 pb:12 mx-auto">
+      {error && (
+        <Alert
+          data-testid="login-error"
+          color="failure"
+          icon={HiInformationCircle}
+        >
+          <span className="font-medium">Auth Error!</span> {parsedError()}
+        </Alert>
+      )}
 
-        {isSignup && (
-          <div>
-            <div className="mb-2 block">
-              <Label htmlFor="password2" value="Confirm password" />
-            </div>
-            <TextInput
-              id="password2"
-              data-testid="login-confirm-pswd"
-              type="password"
-              required
-            />
-          </div>
-        )}
-        <Button data-testid="login-submit-btn" type="submit">
-          Submit
+      {props?.providers?.['credentials'] && isSignup ? (
+        <SignUpForm onSubmit={handleSignUp} />
+      ) : (
+        <SignInForm onSubmit={handleSignIn} />
+      )}
+      {props?.providers?.google && (
+        <Button
+          onClick={() => signIn('google', { redirect: true, callbackUrl: '/' })}
+          outline
+          className="mt-4"
+        >
+          <Image
+            height="24"
+            width="24"
+            alt="Google Provider"
+            id="provider-logo"
+            src="https://authjs.dev/img/providers/google.svg"
+          />
+          <span>Sign in with Google</span>
         </Button>
-      </form>
+      )}
+      <p className="mt-4 dark:text-white">
+        {isSignup ? (
+          <>
+            Already have an account?{' '}
+            <Link href="/login?signup=0">
+              <span className="text-blue-600 hover:underline">Sign in</span>
+            </Link>
+          </>
+        ) : (
+          <>
+            {' '}
+            Don&apos;have an account?{' '}
+            <Link href="/login?signup=1">
+              <span className="text-blue-600 hover:underline">Sign up</span>
+            </Link>
+          </>
+        )}
+      </p>
     </div>
   );
 };
```

Comment:
In case we cut registration logic out of nextauth callback, we can separate singin/signup pages and reduce amount of checks. Code will be much easier to read and will be more predictive.

Reply:
I also planned to create two routes, but I reread the requirements. Unfortunately we are not allowed to split login page into Sign In and Sign Up. According to the task requirements, this should be a single page. This can lead to point decrease during crosscheck.